irohad/model/converters/json_query_factory.hpp

```diff
@@ -48,40 +42,54 @@ namespace iroha {
          * @param query_json string representation of query
          * @return deserialized query
          */
-        optional_ptr <Query> deserialize(const std::string query_json);
+        optional_ptr<Query> deserialize(const std::string query_json);
 
         /**
          * Convert model Query to json string
          * @param model_query - model representation of query
          * @return serialized Query in json format
          */
-        nonstd::optional<std::string> serialize(
-            std::shared_ptr<Query> model_query);
+        std::string serialize(std::shared_ptr<Query> model_query);
 
        private:
-        using Deserializer = std::shared_ptr<Query> (JsonQueryFactory::*)(
-            GenericValue<UTF8<char>>::Object &);
+        Convert<std::shared_ptr<Query>> toQuery;
+
+        optional_ptr<Query> deserialize(const rapidjson::Document &document);
+
+        using Deserializer =
+            optional_ptr<Query> (JsonQueryFactory::*)(const rapidjson::Value &);
 
         std::unordered_map<std::string, Deserializer> deserializers_;
         // Deserialize handlers
-        std::shared_ptr<Query> deserializeGetAccount(GenericValue<UTF8<char>>::Object &obj_query);
-        std::shared_ptr<Query> deserializeGetSignatories(GenericValue<UTF8<char>>::Object &obj_query);
-        std::shared_ptr<Query> deserializeGetAccountTransactions(GenericValue<UTF8<char>>::Object &obj_query);
-        std::shared_ptr<Query> deserializeGetAccountAssetTransactions(GenericValue<UTF8<char>>::Object &obj_query);
-        std::shared_ptr<Query> deserializeGetAccountAssets(GenericValue<UTF8<char>>::Object &obj_query);
+        optional_ptr<Query> deserializeGetAccount(
```

Comment:
Similar with serializiers can be generic, e.g `deserializeQuery<GetAccount>(..)`

Reply:
Out of scope of this pull request.

Reply:
I just asked to mark with todo, that's fine